doc/memory.rst

```diff
@@ -345,6 +346,14 @@ Gotchas
         def method(self, ...):
 	    pass
 
+  Note that the first argument (`self`) will **not** be checked as
+  part of the caching. In other words, if you have two different
+  instances `foo1`, `foo2` which are both of type `Foo` and you call
+  `foo2.method` after `foo1.method` with the same arguments then the
+  second call will return the previously cached result, even though
+  `foo1` and `foo2` are different objects.
```

Comment:
I see where the confusion comes from but I don't think your explanation quite cuts it. The main reason is that you do `self.method = mem.cache(self.method)` you have no self in the arguments any more. So it doesn't make sense that "the first argument ('self') will not be checked for caching".

Maybe we should just say something like the @mem.cache decorator should be used preferably with functions, and inside a class only on staticmethods (or methods that don't use self).

Maybe a simple example would help understand this caveat as well.

@GaelVaroquaux @ogrisel, what do you think?